scripts/generate_reaction_names.py

```diff
@@ -0,0 +1,165 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Generate reaction names.
+
+Reads all reaction identifiers from the metanetx source files, and tries to look
+up the common names for the reaction from BiGG, kegg, modelseed or EC.
+"""
+
+from collections import defaultdict
+import csv
+import logging
+import logging.config
+import json
+import re
+
+from tqdm import tqdm
+import requests
+
+from metanetx import data
+
+logging.config.dictConfig(
+    {
+        "version": 1,
+        "disable_existing_loggers": False,
+        "formatters": {
+            "simple": {
+                "format": "%(asctime)s %(name)s::%(funcName)s:%(lineno)d %(message)s"
+            }
+        },
+        "handlers": {
+            "console": {
+                "level": "DEBUG",
+                "class": "logging.StreamHandler",
+                "formatter": "simple",
+            }
+        },
+        "loggers": {
+            "urllib3.connectionpool": {
+                "level": "INFO",
+                "class": "logging.StreamHandler",
+                "formatter": "simple",
+            }
+        },
+        "root": {"level": "DEBUG", "handlers": ["console"]},
+    }
+)
+logger = logging.getLogger(__name__)
+
+logger.info("Downloading ModelSEED reaction database")
+# ModelSEED ID -> ModelSEED name
+model_seed_map = {}
+response = requests.get(
+    "https://raw.githubusercontent.com/ModelSEED/ModelSEEDDatabase/dev/Biochemistry/reactions.tsv",
+    stream=True,
+)
+response.raise_for_status()
+for row in csv.DictReader(response.iter_lines(decode_unicode=True), delimiter="\t"):
+    model_seed_map[row["id"]] = row["name"]
+logger.info(f"Mapped {len(model_seed_map)} ModelSEED reactions")
+
+logger.info("Reading metanetx source files")
+data.load_metanetx_data()
+
+
+def lookup_bigg(bigg_id):
+    response = requests.get(
+        f"http://bigg.ucsd.edu/api/v2/universal/reactions/{bigg_id}"
+    )
+    response.raise_for_status()
+    return response.json().get("name")
+
+
+def lookup_seed(seed_id):
+    return model_seed_map[seed_id]
+
+
+def lookup_kegg(kegg_id):
+    response = requests.get(f"http://rest.kegg.jp/find/reaction/{kegg_id}")
+    response.raise_for_status()
+    line = response.text
+    # The line format wasn't well documented, splitting on tab and semicolon was
+    # just found by inspecting a couple of examples. Might not hold up
+    # perfectly.
+    return line.split("\t", 1)[1].split(";", 1)[0]
+
+
+def lookup_ec(ec):
+    # Try to look up name based on EC number
+    # Only try if the EC numbers is exact (4 numbers), and there is only a
+    # single one.
+    regex = r"^\d+\.\d+\.\d+\.\d+$"
+    if re.match(regex, ec):
+        response = requests.get(f"https://enzyme.expasy.org/EC/{ec}.txt")
+        response.raise_for_status()
+        for line in response.text.split("\n"):
+            key, value = line.split(maxsplit=1)
+            if key == "DE":
```

Comment:
I disagree with only collecting one name. We should aggregate all the different synonyms for reactions such that a user might find a reaction through various possible names.

Reply:
Sounds like a good idea, I've created https://github.com/DD-DeCaF/metanetx/issues/4